styles/main.css

```diff
@@ -84,6 +84,9 @@ ul {
     -moz-box-shadow: 4px 4px 4px rgba(255, 0, 0, 0.3);
     -webkit-box-shadow: 4px 4px 4px rgba(255, 0, 0, 0.3);
     box-shadow: 4px 4px 4px rgba(255, 0, 0, 0.3);
+
+    display: flex;
+    flex-direction: column;

```

Comment:
You need these two declarations to make the footer's auto margin-top have the desired effect (extra space in flex layouts may get distributed to auto margins)